core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -58,8 +60,11 @@
         ConfigFactory.invalidateCaches();
         // The thread's context class loader will be used for application classpath resources,
         // while the driver class loader will be used for reference classpath resources.
-        return ConfigFactory.defaultApplication()
-            .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader()))
+        return ConfigFactory.defaultOverrides()
+            .withFallback(
+                ConfigFactory.defaultApplication()
+                    .withFallback(defaultReference(CqlSession.class.getClassLoader())))
```

Comment:
Nit: withFallback calls can be chained, this is more readable. I'll fix it directly as part of the merge.